vscode/src/circuitEditor.ts

```diff
@@ -0,0 +1,169 @@
+          return;
+        case "update":
+          this.updateTextDocument(document, e.text);
+          return;
+        case "read":
+          updateWebview();
+          return;
+      }
+    });
+
+    const updateWebview = () => {
+      const circuit = this.getDocumentAsJson(document);
+      const filename = document.fileName.split(/\\|\//).pop()!.split(".")[0];
+
+      const props = {
+        title: `${filename} Circuit`,
+        targetProfile: "",
+        simulated: false,
+        calculating: false,
+        circuit,
+      };
+
+      const message = {
+        command: "circuit",
+        props,
+      };
+      webviewPanel.webview.postMessage(message);
+    };
+
+    // Update the webview when the text document changes
+    // const changeDocumentSubscription = vscode.workspace.onDidChangeTextDocument(
+    //   (event) => {
+    //     if (event.document.uri.toString() === document.uri.toString()) {
+    //       if (
+    //         this.savedVersion !== undefined &&
+    //         event.document.version > this.savedVersion
+    //       ) {
+    //         // Update the webview with the new document content
+    //         updateWebview();
+    //       }
+    //     }
+    //   },
+    // );
+
+    // // Dispose of the event listener when the webview is closed
+    // webviewPanel.onDidDispose(() => {
+    //   changeDocumentSubscription.dispose();
+    // });
+
+    updateWebview();
+  }
+
+  private getHtmlForWebview(webview: vscode.Webview): string {
+    const extensionUri = this.context.extensionUri;
+
+    function getUri(pathList: string[]) {
+      return webview.asWebviewUri(
+        vscode.Uri.joinPath(extensionUri, ...pathList),
+      );
+    }
+
+    const katexCss = getUri(["out", "katex", "katex.min.css"]);
+    const githubCss = getUri(["out", "katex", "github-markdown-dark.css"]);
+    const webviewCss = getUri(["out", "webview", "webview.css"]);
+    const scriptUri = getUri(["out", "webview", "editor.js"]);
+    const resourcesUri = getUri(["resources"]);
+    return `
+      <!DOCTYPE html>
+      <html lang="en">
+        <head>
+          <meta charset="UTF-8">
+          <meta name="viewport" content="width=device-width, initial-scale=1.0">
+          <title>Q#</title>
+          <link rel="stylesheet" href="${githubCss}" />
+          <link rel="stylesheet" href="${katexCss}" />
+          <link rel="stylesheet" href="${webviewCss}" />
+          <script src="${scriptUri}"></script>
+          <script>
+            window.resourcesUri = "${resourcesUri.toString()}";
+          </script>
+        </head>
+        <body>
+        </body>
+      </html>`;
+  }
+
+  private getDocumentAsJson(document: vscode.TextDocument): any {
+    const text = document.getText();
+    if (text.trim().length === 0) {
+      return {};
+    }
+
+    try {
+      return JSON.parse(text);
+    } catch {
+      throw new Error(
+        "Could not get document as json. Content is not valid json",
```

Comment:
How is this error surfaced to the user? Does someone catch this? 

This is what I see in the UI if the JSON is corrupt:

![image](https://github.com/user-attachments/assets/2d09ccb7-d946-40c7-a2fa-d8e3546bd865)

Reply:
Okay I've changed the logic a bit so that the error stops the circuit from rendering and displays the error message instead:
![image](https://github.com/user-attachments/assets/93aeee96-a0a6-47f6-acb4-5d901d473fe7)